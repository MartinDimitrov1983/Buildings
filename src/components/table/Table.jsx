import React from 'react';
import TableRow from './TableRow.jsx';
import styles from './index.module.css';
import { HEADER_VALUES } from '../../utils/constants.js';

const Table = ({ data, onDelete, ...props }) => {
    return (
        <div className={styles.container} {...props}>
            <table className={styles.table}>
                <thead>
                    <TableRow
                        tableCells={HEADER_VALUES}
                        tag="th"
                        isBodyRow={false}
                    />
                </thead>
                <tbody>
                    {data?.map((rowData) => (
                        <TableRow
                            key={`key-${rowData.id}-${rowData.name}`}
                            tableCells={rowData}
                            tag="td"
                            onDelete={onDelete}
                        />
                    ))}
                </tbody>
            </table>
        </div>
    );
};

export default Table;
