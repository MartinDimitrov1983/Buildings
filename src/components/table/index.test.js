import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import Table from './Table';

test('Snapshot Table', () => {
    const { asFragment } = render(
        <BrowserRouter>
            <Table />
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});
