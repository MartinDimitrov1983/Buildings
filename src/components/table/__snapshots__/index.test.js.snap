// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Table 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            Id
          </th>
          <th>
            Name
          </th>
          <th>
            Area
          </th>
          <th>
            Location
          </th>
          <th>
            Image
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</DocumentFragment>
`;
